os/Kconfig

```diff
@@ -198,6 +198,45 @@ config BUILD_KERNEL
 
 endchoice # Build configuration
 
+if BUILD_PROTECTED
+
+config AUTOGEN_MEMORY_LDSCRIPT
+	bool "Auto generate memory.ld linker script"
+	default n
+	---help---
+		Auto generate the board specific memory.ld linker script using values
+		from the configuration file
+
+if AUTOGEN_MEMORY_LDSCRIPT
+
+config FLASH_START_ADDR
```

Comment:
It looks little strange on relationship between autogen and memory information.
And flash_start_addr is necessary to run download script always.

Reply:
OK. I removed the AUTOGEN_MEMORY_LDSCRIPT condition.